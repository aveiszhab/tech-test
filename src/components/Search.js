import React from 'react';
import '../styles/Search.css';

const Search = () => {
    
    return(
        <>
            <form
                className='search-form'
            >
                <input
                    className='search-input'
                    type='text'
                    placeholder='Search for an image'
                >
                </input>
                <button
                    className='submit-button'
                    type='submit'
                    name='go'
                >
                    Go!
                </button>
            </form>
        </>
    )   
};

export default Search;
