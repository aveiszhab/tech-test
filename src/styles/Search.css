.search-form {
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin-bottom: 10%;
}

.search-input {
    width: 20vw;
    height: 2vw;
    border: 1px solid gray;
    border-radius: 4px 0px 0px 4px;
    font-size: 1.5vw;
    padding: 0%;
}

button {
    width: 5vw;
    border-radius: 0px 4px 4px 0px;
    background-color: rgb(228, 35, 35);
    border: none;
    font-size: 1.5vw;
    padding: 0%;
    color: white;
    
}
