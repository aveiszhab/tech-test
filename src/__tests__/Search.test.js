import React from 'react';
import { render, screen } from '@testing-library/react';
import Search from '../components/Search';

describe('Search', () => {
    it('renders correctly to match the Search snapshot', () => {
        const {asFragment} = render(<Search />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders form correctly', () => {
        render(<Search />);

        const searchBox = screen.getByRole('textbox', {placeholder:'Search for an image'});
        expect(searchBox).toBeInTheDocument();
        expect(searchBox).toHaveClass('search-input');
        expect(searchBox).toHaveAttribute('type', expect.stringContaining('text'));

        const goButton = screen.getByRole('button', {name: /go/i});
        expect(goButton).toHaveClass('submit-button');
        expect(goButton).toBeEnabled();
        expect(goButton).toHaveAttribute('type','submit');
        expect(goButton).toHaveTextContent(/go/i)
    });
});