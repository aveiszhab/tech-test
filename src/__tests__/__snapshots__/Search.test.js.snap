// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly to match the Search snapshot 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-input"
      placeholder="Search for an image"
      type="text"
    />
    <button
      class="submit-button"
      name="go"
      type="submit"
    >
      Go!
    </button>
  </form>
</DocumentFragment>
`;
